SignalServiceKit/src/Loki/Utilities/FriendRequestExpireJob.swift

```diff
@@ -0,0 +1,214 @@
+/*
+ This class is used for settings friend requests to expired
+ This is modelled after `OWSDisappearingMessagesJob`.
+ */
+
+@objc(OWSLokiFriendRequestExpireJob)
+public class FriendRequestExpireJob: NSObject {
+    
+    private let databaseConnection: YapDatabaseConnection
+    private let messageFinder: FriendRequestExpireMessageFinder
+    
+    // These three properties should only be accessed on the main thread.
+    private var hasStarted = false
+    private var fallbackTimer: Timer?
+    private var nextExpireTimer: Timer?
+    private var nextExpireDate: Date?
+    
+    // Our queue
+    public static var serialQueue: DispatchQueue = {
```

Comment:
Why not just `public static let serialQueue = DispatchQueue(label: "network.loki.friendrequest.expire")`?

Reply:
Plus this should be `fileprivate`, right?